sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionBasedLoadBalancer.java

```diff
@@ -286,10 +286,101 @@ private Mono<Void> loadBalance(final Tuple2<Map<String, PartitionOwnership>, Lis
+                Map<String, PartitionOwnership> partitionOwnershipMap = tuple.getT1();
+
+                List<String> partitionIds = tuple.getT2();
+
+                if (CoreUtils.isNullOrEmpty(partitionIds)) {
+                    // This may be due to an error when getting Event Hub metadata.
+                    throw logger.logExceptionAsError(Exceptions.propagate(
+                        new IllegalStateException("There are no partitions in Event Hub " + eventHubName)));
+                }
+                partitionsCache.set(partitionIds);
+
+                if (!isValid(partitionOwnershipMap)) {
+                    // User data is corrupt.
+                    throw logger.logExceptionAsError(Exceptions.propagate(
+                        new IllegalStateException("Invalid partitionOwnership data from CheckpointStore")));
+                }
+
+                /*
+                 * Remove all partitions' ownership that have not been modified for a configuration period of time. This
+                 * means that the previous EventProcessor that owned the partition is probably down and the partition is now
+                 * eligible to be claimed by other EventProcessors.
+                 */
+                Map<String, PartitionOwnership> activePartitionOwnershipMap = removeInactivePartitionOwnerships(
+                    partitionOwnershipMap);
+
+                /*
+                 * If there are active partition ownership, this means the checkpoint store claimed partition successfully,
+                 * we don't need to check it.
+                 */
+                if (!CoreUtils.isNullOrEmpty(activePartitionOwnershipMap)) {
+                    return Mono.empty();
+                }
+
+                /*
+                 * If there is no active partition ownership, claim a random partition and the other partitions will be
+                 * claimed in load balance.
+                 */
+                PartitionOwnership ownershipRequest = createPartitionOwnershipRequest(partitionOwnershipMap,
+                    partitionIds.get(RANDOM.nextInt(partitionIds.size())));
+                List<PartitionOwnership> partitionToClaim = Collections.singletonList(ownershipRequest);
+                return this.checkpointStore.claimOwnership(partitionToClaim)
+                    .collectList()
+                    .zipWhen(ownershipList -> checkpointStore.listCheckpoints(fullyQualifiedNamespace, eventHubName,
+                            consumerGroupName)
+                        .collectMap(checkpoint -> checkpoint.getPartitionId(), Function.identity()))
+                    .flatMap(ownedPartitionCheckpointsTuple -> {
+                        ownedPartitionCheckpointsTuple.getT1()
+                            .stream()
+                            .forEach(po -> partitionPumpManager.startPartitionPump(po,
+                                ownedPartitionCheckpointsTuple.getT2().get(po.getPartitionId())));
```

Comment:
Why do we startPartitionPump here?

Reply:
If we don't startPartitionPump here, the partition won't start pump until elapse expirationInterval (default value is 2 minutes). So here are three solutions:

1. Just claim partition, don't start pump. This partition wil start pump after expirationInterval.
2. Claim partition and set an early lastModifiedTime, then loadbalance can start this partition quickly, I try to avoid modifying the `claimOwnerShip` method.
3. Claim partion and start pump.

I choose the third solution, actually I'm not sure which one is the best.

Reply:
Firstly, claim ownerShip and start pump are at same time. Before this PR, loadbalance will find unclaimed partition or inactive partition(claimed but didn't get modify for expirationInterval) to claim and start pump.

Reply:
OK, make sense.

So before this PR, all partitions is claimed and started by load balancer.
After this PR, 1 partition is claimed and started here. And the rest will be claimed and started by load balancer?

Reply:
yeah